Synced with template and updated box version 0.6.0

claesgill opened the pull request:
Closes #34 

- Synced with template
- Updated box version 0.4.0 -> 0.6.0

zhenik committed:
Add terraform-provider-vault token (#36)

Neha-Sinha2305 committed:
Updated makefile targets up and template_example

Neha-Sinha2305 committed:
Merge pull request #39 from fredrikhgrelland/update_makefile_ci_improv

Updated makefile targets up and template_example

zhenik committed:
Add doc sync module with the latest template AND pre-commit fix (#40)

* update documentation

* update pre-commit target

dangernil committed:
add changelog enforcer (#42)

* added changelog enforcer

dangernil committed:
Automate release process (#43)

* automate the release process

Neha-Sinha2305 committed:
Updated makefile targets up and template_example

Neha-Sinha2305 committed:
Fixed workflow file

Neha-Sinha2305 committed:
Fixed linter issue

Neha-Sinha2305 committed:
Merge pull request #46 from fredrikhgrelland/changes_for_test_run_gitlab

Updated makefile targets up and template_example

claesgill committed:
Bumped vagrant version (#51)

* Bumped vagrant version. 
* Removed extra lines in Makefile
* Added boot timeout

fredrikhgrelland committed:
redesign to fit hashistack 0.6.0 (#44)

* redesign to fit hashistack 0.5.0

* move to 0.6.0

pdmthorsrud committed:
Added getting started section (#49)

* Added getting started section

Added more headers

Added more sections. Sorry, I lost count...

Moved getting started to separate file

removing all fluff, got about halfway, nighty!

added some newlines here and there

added a modules getting started

ADDED vagrantbox getting started

changed to ordered TOC

* formatting and small fixes
Co-authored-by: Claes Gill <claes@claesgill.com>
Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

* Filled out section 6

* small fixes

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>
Co-authored-by: Claes Gill <claes@claesgill.com>

* Added CICD section

* fixed last todos
Co-authored-by: Claes Gill <claes@claesgill.com>

* Formatting and small changes

Co-authored-by: Claes Gill <claes@claesgill.com>
Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

claesgill committed:
Synced with template and updated box version 0.6.0

zhenik commented on dev/vagrant/conf/README.md:
```diff
@@ -10,7 +10,7 @@ There are two layers of configuration built into the box.
 
 This might come handy if you need to change or replace that bootstrap process. For example you replacing the entire `/etc/ansible` directory.
-For most cases, you are probably looking to add configuration in [pre/poststart bootstrap](pre_bootstrap/README.md)
+For most cases, you are probably looking to add configuration in [pre/poststart bootstrap](../pre_bootstrap/README.md)
```
```suggestion
For most cases, you are probably looking to add configuration in [pre/pre_bootstrap](../pre_bootstrap/README.md)
```

claesgill replied:
These should probably  be fixes in the template aswell

zhenik commented on template_example/dev/vagrant/bootstrap/README.md:
```diff
@@ -0,0 +1,15 @@
+# Customize the startup procedure of the Hashistack with pre- and post start ansible scripts
+
+You may put any number of script files in this directory for running ansible commands prior to bootstrapping the hashistack.
+The bootstrap procedure is included/hardcoded in your box.
```
```suggestion
The bootstrap procedure is included/hardcoded in vagrant-hashistack box.
```

claesgill replied:
These should probably  be fixes in the template aswell

claesgill committed:
Update dev/vagrant/conf/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Update template_example/dev/vagrant/bootstrap/README.md

Co-authored-by: Nikita Zhevnitskiy <nikita.zhevnitskiy@gmail.com>

Renmaed file to change execution order/sequence

Moved file under Vault folder

Adding use of vault token to ansible-playbook

New task that set vault token

Added lookup in 01 and 011 files

Added consul token

Should work :shrug:

Adding debug output

Added debug tokens

Added envs

Testing with Vault provider

Adding env vars for ansible tasks

Testing :shrug:

Upd ansible playbooks

:shrug::rocket:

Removed use of token when acl disabled

claesgill commented:
Should [022-update-nomad-write-default-policy.yml](dev/vagrant/bootstrap/nomad/post/022-update-nomad-write-default-policy.yml) be named 030 instead of 022 as per execution sequence?

zhenik commented:
> Regarding the execution sequence:
> 
> Should [00-create_dir_for_persisting_data.yml](dev/vagrant/bootstrap/nomad/pre/00-create_dir_for_persisting_data.yml) and [022-update-nomad-write-default-policy.yml](dev/vagrant/bootstrap/nomad/post/022-update-nomad-write-default-policy.yml) be named 030 and 031 instead of 00 and 022?

these numbers just to define seq. And try to have some room, you might need in the future. So I recommend `030` and `040`

zhenik commented on dev/ansible/01_generate_secrets_vault.yml:
```diff
@@ -1,9 +1,11 @@
   run_once: true
   environment:
-      VAULT_ADDR: "http://127.0.0.1:8200"
-      VAULT_TOKEN: $vault_master_token
```
you might need to make a lookup, here is example https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/ansible/tests/enterprise/vault_namespaces_test.yml#L16

claesgill replied:
This is outdated - created a new task instead that exports `VAULT_TOKEN`. But should probably use lookup anyways :smiley:

zhenik reviewed (dismissed):
Tough one. Good job
